cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java

```diff
@@ -239,12 +257,23 @@ public void replicatorBuilderAddsCookieInterceptorDefaultPort() throws Exception
                         getRootUri()).
                         toString()
         );
-        // we know there will only be one interceptor and it will be first in the list
-        Assert.assertEquals(CookieInterceptor.class,
-                ((BasicPullStrategy)r.strategy).requestInterceptors.get(0).getClass());
-        Assert.assertEquals(CookieInterceptor.class,
-                ((BasicPullStrategy)r.strategy).responseInterceptors.get(0).getClass());
-        // TODO how to assert user/pass in cookie interceptor is correct
+        // peek inside these private fields to see that interceptors have been set
+        Field reqIField = ReplicatorBuilder.class.getDeclaredField("requestInterceptors");
```

Comment:
I think we should refactor this code to be a helper method, so we can use it in `replicatorBuilderAddsCookieInterceptorCustomPort` too